### PR TITLE
dotCMS/core#24339 [UI] Open the traffic split sidebar from the Variants traffic indicator. 

### LINKED ISSUE #24339: [UI] Open the traffic split sidebar from the Variants traffic indicator.
### Parent Issue

_No response_

### Task

As a user I want to be able to edit the traffic split from the variant traffic indicator. 

image

### Proposed Objective

User Experience

### Proposed Priority

Priority 3 - Average

### Acceptance Criteria

The sidebar should open when click the traffic indication in the variant description. 
Refactor Variants step to follow the same behavior as other steps. 

### External Links... Slack Conversations, Support Tickets, Figma Designs, etc.

_No response_

### Assumptions & Initiation Needs

_No response_

### Quality Assurance Notes & Workarounds

_No response_

### Sub-Tasks & Estimates

_No response_

### PR DESCRIPTION
**Pull request by hmoreras**
### Proposed Changes
* Allow users to change the traffic weight from the variant section in the configuration screen 
* Refactor variants section to follow same patter as other steps. 

### Screenshots

https://user-images.githubusercontent.com/31667212/225441284-8aa78fc2-029d-4683-85cf-5d30c166d27d.mov

### CODE REVIEW

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.ts [hunk 1]
```diff
@@ -65,7 +65,7 @@ export class DotExperimentsConfigurationSchedulingComponent {
     }
 
     private handleSidebar(status: StepStatus) {
-        if (status && status.isOpen) {
+        if (status && status.isOpen && status.status != ComponentStatus.SAVING) {
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.ts:68]:
```suggestion
        if (this.shouldLoadSidebar(status)) {
```

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts [hunk 2]
```diff
@@ -85,31 +78,27 @@ export class DotExperimentsConfigurationTrafficComponent {
...
-        }
+        this.sidebarHost.viewContainerRef.clear();
+        this.componentRef =
+            status.experimentStep == ExperimentSteps.TRAFFICS_SPLIT
+                ? this.sidebarHost.viewContainerRef.createComponent<DotExperimentsConfigurationTrafficSplitAddComponent>(
+                      DotExperimentsConfigurationTrafficSplitAddComponent
+                  )
+                : this.sidebarHost.viewContainerRef.createComponent<DotExperimentsConfigurationTrafficAllocationAddComponent>(
+                      DotExperimentsConfigurationTrafficAllocationAddComponent
+                  );
```
Comment (fmontes) [inline core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts:101]:
```suggestion
                   const CurrentComponent = this.status.experimentStep == ExperimentSteps.TRAFFICS_SPLIT ? DotExperimentsConfigurationTrafficSplitAddComponent : DotExperimentsConfigurationTrafficAllocationAddComponent;

        this.sidebarHost.viewContainerRef.createComponent<CurrentComponent>(CurrentComponent)
```

Comment (fmontes) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts:101]: Not sure tho.

**Review by fmontes — DISMISSED** [review]

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts (hunk 2 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-traffic/dot-experiments-configuration-traffic.component.ts:101]: in the type wont work.

File: core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.ts (hunk 1 above)
Comment (hmoreras) [reply core-web/apps/dotcms-ui/src/app/portlets/dot-experiments/dot-experiments-configuration/components/dot-experiments-configuration-scheduling/dot-experiments-configuration-scheduling.component.ts:68]: done

**Review by oidacra — APPROVED** [review]

**Review by freddyucv — APPROVED** [review]

**Pull request merged**